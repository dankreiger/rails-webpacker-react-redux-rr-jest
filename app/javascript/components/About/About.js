import React from 'react';
import { AboutContainer } from './About.styles';

const About = () => {
  return <AboutContainer>This is about...ain't nothing here B.</AboutContainer>;
};

export default About;
