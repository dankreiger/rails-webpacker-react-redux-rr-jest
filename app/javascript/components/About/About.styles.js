import styled from 'styled-components';
import { Container } from 'reactstrap';

export const AboutContainer = styled(Container)`
  display: flex;
  height: 100vh;
  justify-content: center;
  align-items: center;
`;
