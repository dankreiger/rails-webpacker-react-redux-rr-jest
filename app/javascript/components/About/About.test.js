import About from './About';
import { shallow } from 'enzyme';

describe('About', () => {
  let aboutComponent;

  beforeEach(() => {
    aboutComponent = shallow(<About />);
  });

  describe('rendering', () => {
    test('renders as expected', () => {
      expect(aboutComponent.length).toBe(1);
      expect(aboutComponent).toMatchSnapshot();
    });
  });
});
