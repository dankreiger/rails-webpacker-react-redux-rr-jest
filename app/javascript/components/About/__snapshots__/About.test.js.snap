// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About rendering renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <About />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "This is about...ain't nothing here B.",
    },
    "ref": null,
    "rendered": "This is about...ain't nothing here B.",
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  height: 100vh;
  justify-content: center;
  align-items: center;
",
        ],
      },
      "displayName": "Styled(Container)",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "This is about...ain't nothing here B.",
      },
      "ref": null,
      "rendered": "This is about...ain't nothing here B.",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  height: 100vh;
  justify-content: center;
  align-items: center;
",
          ],
        },
        "displayName": "Styled(Container)",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
